models/DataObject/Data/QuantityValue.php

```diff
@@ -104,6 +104,42 @@ public function getValue()
         return $this->value;
     }
 
+    /**
+     * @param Unit $toUnit
+     *
+     * @return QuantityValue
+     * @throws \Exception
+     */
+    public function convert(Unit $toUnit) {
+        $fromUnit = $this->getUnit();
+        if(!$fromUnit instanceof Unit) {
+            throw new \Exception('Quantity value has no unit');
+        }
+
+        $fromBaseUnit = $fromUnit->getBaseunit();
+        if($fromBaseUnit === null) {
+            $fromUnit = clone $fromUnit;
+            $fromBaseUnit = $fromUnit;
+            $fromUnit->setFactor(1);
+            $fromUnit->setConversionOffset(0);
+        }
+
+        $toBaseUnit = $toUnit->getBaseunit();
+        if($toBaseUnit === null) {
+            $toUnit = clone $toUnit;
+            $toBaseUnit = $toUnit;
+            $toUnit->setFactor(1);
+            $toUnit->setConversionOffset(0);
+        }
+
+        if($fromBaseUnit === null || $toBaseUnit === null || $fromBaseUnit->getId() !== $toBaseUnit->getId()) {
```

Comment:
is this even possible ?

$toBaseUnit === null

Reply:
Of course this is possible. When the target unit does not have a base unit.
For example when you have an object with a quantity value in unit "cm". 
And you have configured units m, cm and mm. base unit of cm and mm is m. m does not have a base unit.
When you want to convert cm to m, toUnit is m and it does not have a base unit. Then it is assumed that the target unit has itself as base unit.